test/test/lock/LockTests.java

```diff
@@ -10,6 +10,40 @@
 import one.profiler.test.TestProcess;
 
 public class LockTests {
+    private static void contendedLocks(TestProcess p, int threshold, double minRatio) throws Exception {
+        Output out = p.profile("-e lock --lock " + threshold + " --threads -o collapsed");
+        Output waitingLocks = out.filter("LockProfiling\\.contend;java\\.lang\\.Object");
+
+        double ratio = 100 * out.ratio("contend-0.5-", "contend-0.05-");
+
+        System.out.println("threshold = " + threshold + ", ratio = " + ratio + ", minRatio = " + minRatio);
+
+        assert (ratio > minRatio) || (Double.isNaN(ratio) && Double.isNaN(minRatio));
+    }
+
+    // 0 is equivalent to disabling sampling of locks, so all profiles are included.
+    @Test(mainClass = LockProfiling.class, inputs = { 0, 75 })
+    @Test(mainClass = LockProfiling.class, inputs = { 1, 75 })
+    @Test(mainClass = LockProfiling.class, inputs = { 10, 75 })
+    @Test(mainClass = LockProfiling.class, inputs = { 100, 75 })
+    @Test(mainClass = LockProfiling.class, inputs = { 1000, 75 })
+    @Test(mainClass = LockProfiling.class, inputs = { 10000, 75 })
```

Comment:
Too many options, IMO - this will significantly increase test run time. I'd reduce the number of tests to the required minimum (let's say, up to 10 seconds in total).